timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala

```diff
@@ -434,27 +434,24 @@ case class TimeSeriesScheduler(logger: Logger) extends Scheduler[TimeSeries] wit
         }
       }
 
-      if (errors.isEmpty) {
-        val executionsToCancel = listExecutionsToCancelGivenBackfills(validBackfills, _state(), runningExecutions.keys)
-        executionsToCancel.foreach { e =>
-          logger.info(s"Cancelling running execution ${e.id} for job ${e.job.id}")
-          e.cancel()
-        }
-        Future.sequence(executionsToCancel.map(runningExecutions.apply))
-          .andThen { case _ => validBackfills.foreach { newBackfill =>
-            updateBackfillState(newBackfill)
-            Database.createBackfill(newBackfill).transact(xa)
-          } }
-      }
-      errors.toList
+      (errors.toList, validBackfills)
     }
 
-
-
     if (backfillErrors.nonEmpty)
       IO.pure(Left(backfillErrors.mkString("\n")))
-    else
+    else {
+      val executionsToCancel = listExecutionsToCancelGivenBackfills(backfillsTodo, _state(), runningExecutions.keys)
```

Comment:
Don't you think that this block should be wrapped in an atomic because it's modifying the state?